src/Workleap.Extensions.Mongo.Tests/MongoIndexerTests.MultipleIndexers.cs

```diff
@@ -0,0 +1,135 @@
+            Assert.Contains("contact_800a8c19f5008939a89b582cc7c22e40", indexNames);
+        }
+    
+        // Simulate a N-1 release with some initial indexes
+        var documentTypes = new[] { typeof(BaseMultipleIndexersTestDocument), typeof(ChildMultipleIndexersTestDocument), typeof(SiblingMultipleIndexersTestDocument) };
+        await this.Services.GetRequiredService<IMongoIndexer>().UpdateIndexesAsync(documentTypes);
+    
+        // Simulate the N release where we remove one indexer and update an other 
+        var updatedDocumentTypes = new[] { typeof(BaseMultipleIndexersTestDocument), typeof(UpdatedChildMultipleIndexersTestDocument) };
+        await this.Services.GetRequiredService<IMongoIndexer>().UpdateIndexesAsync(updatedDocumentTypes);
+        
+        await AssertIndexes<BaseMultipleIndexersTestDocument>();
+    }
+    
+    [BsonKnownTypes(typeof(ChildMultipleIndexersTestDocument))]
+    [MongoCollection("multipleIndexProviders", IndexProviderType = typeof(BaseMultipleIndexersProvider))]
+    private class BaseMultipleIndexersTestDocument : MongoDocument
+    {
+        [BsonElement("name")]
+        public string Name { get; set; } = string.Empty;
+    }
+    
+    private sealed class BaseMultipleIndexersProvider : MongoIndexProvider<BaseMultipleIndexersTestDocument>
+    {
+        public override IEnumerable<CreateIndexModel<BaseMultipleIndexersTestDocument>> CreateIndexModels()
+        {
+            yield return new CreateIndexModel<BaseMultipleIndexersTestDocument>(
+                Builders<BaseMultipleIndexersTestDocument>.IndexKeys.Ascending(x => x.Name),
+                new CreateIndexOptions { Name = "name" });
+        }
+    }
+
+    [BsonDiscriminator("Children")]
+    [MongoCollection("multipleIndexProviders", IndexProviderType = typeof(ChildMultipleIndexersProvider))]
+    private class ChildMultipleIndexersTestDocument : BaseMultipleIndexersTestDocument
+    {
+        [BsonElement("email")]
+        public string Email { get; set; } = string.Empty;
+        
+        [BsonElement("phone")]
+        public string PhoneNumber { get; set; } = string.Empty;
+    }
+    
+    private sealed class ChildMultipleIndexersProvider : MongoIndexProvider<ChildMultipleIndexersTestDocument>
+    {
+        public override IEnumerable<CreateIndexModel<ChildMultipleIndexersTestDocument>> CreateIndexModels()
+        {
+            yield return new CreateIndexModel<ChildMultipleIndexersTestDocument>(
+                Builders<ChildMultipleIndexersTestDocument>
+                    .IndexKeys.Ascending(x => x.Email),
+                new CreateIndexOptions { Name = "contact" });
+        }
+    }
+    
+    // Represent an update in the indexes.
+    [BsonDiscriminator("Children")]
+    [MongoCollection("multipleIndexProviders", IndexProviderType = typeof(UpdatedChildMultipleIndexersProvider))]
+    private sealed class UpdatedChildMultipleIndexersTestDocument : ChildMultipleIndexersTestDocument
```

Comment:
What if this one Inherited from BaseDocument directly instead of  inheriting from ChildDocument?

Reply:
It would be consider like the classic flow no? Like SiblingMultipleIndexersTestDocument or BaseMultipleIndexersTestDocument